src/librustc/infer/region_inference/mod.rs

```diff
@@ -1543,7 +1543,7 @@ impl<'a, 'gcx, 'tcx> GenericKind<'tcx> {
         match *self {
             GenericKind::Param(ref p) => p.to_ty(tcx),
             GenericKind::Projection(ref p) => tcx.mk_projection(
-                p.trait_ref(tcx).clone(), p.item_name(tcx)),
+                p.item_def_id, p.substs),
```

Comment:
I think it fits on one line now, which would look nicer (this applies everywhere).